tests/integration/cht-datasource/person.spec.js

```diff
@@ -0,0 +1,43 @@
+const utils = require('@utils');
+const { Person, Qualifier, getRemoteDataContext} = require('@medic/cht-datasource');
+const personFactory = require('@factories/cht/contacts/person');
+const { expect } = require('chai');
+
+describe('Person', () => {
+  const dataContext = getRemoteDataContext(utils.getOrigin());
+  const personType = 'person';
+  const contact0 = utils.deepFreeze(personFactory.build({ name: 'contact0', role: 'chw' }));
+  const contact1 = utils.deepFreeze(personFactory.build({ name: 'contact1', role: 'chw' }));
+  const contact2 = utils.deepFreeze(personFactory.build({ name: 'contact2', role: 'chw' }));
+  const contact3 = utils.deepFreeze(personFactory.build({ name: 'contact3', role: 'chw' }));
+  const allDocs = [contact0, contact1, contact2, contact3];
+  const e2eTestUser = {
+    '_id': 'e2e_contact_test_id',
+    'type': 'person',
+  };
+
+  before(async () => {
+    await utils.saveDocs(allDocs);
+  });
+
+  after(async () => {
+    await utils.revertDb([], true);
+  });
+
+  describe('v1', async () => {
+    describe('getAll', async () => {
```

Comment:
Good test! Since this is technically still just calling `GET /api/v1/person` underneath the hood, I would vote we just add this test case into the `'GET /api/v1/person'` describe block of the `controllers/person.spec.ts` file (next to the rest of the new test cases).  It is true that it is one more step removed from the controller, but it really is not actually adding any additional functionality besides abstracting away the paging logic of the REST calls...

If we had a bunch of other tests to add here that did not really fit anywhere else, that would be one thing, but as things stand now, I think we can just keep it simple.  Happy to discuss more if you all see it differently.  @sugat009 @tatilepizs

Reply:
I'm ok with any as technically it's still calling the API under the hood, however, to keep things separated the current one would be the way to go but then again even for `Person.v1.get` or `Person.v1.getPage` we technically aren't directly calling the REST API endpoints so moving this also makes sense.
Let's move this test there for now and if things tend to branch out more we'll move it back out.